src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -177,32 +202,36 @@ impl<Src: OpendalSource> OpendalReader<Src> {
                 // EOF
                 break;
             }
+            let msg_offset = (offset + n_read).to_string();
             // note that the buffer contains the newline character
             debug_assert_eq!(n_read, line_buf.len());
+            if (object_name.ends_with(".gz") || object_name.ends_with(".gzip"))
+                && offset + n_read <= start_offset
+            {
+                // For gzip compressed files, the reader needs to read from the beginning each time,
+                // but it needs to skip the previously read part and start yielding chunks from a position greater than or equal to start_offset.
+            } else {
+                batch.push(SourceMessage {
+                    key: None,
+                    payload: Some(std::mem::take(&mut line_buf).into_bytes()),
+                    offset: msg_offset,
+                    split_id: split.id(),
+                    meta: SourceMeta::Empty,
+                });
+            }
 
-            // FIXME(rc): Here we have to use `offset + n_read`, i.e. the offset of the next line,
-            // as the *message offset*, because we check whether a file is finished by comparing the
-            // message offset with the file size in `FsFetchExecutor::into_stream`. However, we must
-            // understand that this message offset is not semantically consistent with the offset of
-            // other source connectors.
-            let msg_offset = (offset + n_read).to_string();
-            batch.push(SourceMessage {
-                key: None,
-                payload: Some(std::mem::take(&mut line_buf).into_bytes()),
-                offset: msg_offset,
-                split_id: split.id(),
-                meta: SourceMeta::Empty,
-            });
```

Comment:
Sorry I just see this change a long time after this PR got merged. Want to confirm that, is the `msg_offset` for plain files (not compressed) still the "end offset" of a line? If so then I think it is still not intuitive, and still need a comment here.